### PR TITLE
8337958: Out-of-bounds array access in secondary_super_cache

### PR DESCRIPTION
**Pull request by theRealAph**
Hi all,

This pull request contains a backport of commit [03204600](https://github.com/openjdk/jdk/commit/03204600c596214895ef86581eba9722f76d39b3) from the [openjdk/jdk](https://git.openjdk.org/jdk) repository.

The commit being backported was authored by Andrew Haley on 12 Aug 2024 and was reviewed by Vladimir Ivanov and Aleksey Shipilev.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8337958](https://bugs.openjdk.org/browse/JDK-8337958) needs maintainer approval

### Issue
 * [JDK-8337958](https://bugs.openjdk.org/browse/JDK-8337958): Out-of-bounds array access in secondary_super_cache (**Bug** - P3 - Approved)

### Reviewers
 * [Vladimir Ivanov](https://openjdk.org/census#vlivanov) (@iwanowww - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk23u.git pull/63/head:pull/63` \
`$ git checkout pull/63`

Update a local copy of the PR: \
`$ git checkout pull/63` \
`$ git pull https://git.openjdk.org/jdk23u.git pull/63/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 63`

View PR using the GUI difftool: \
`$ git pr show -t 63`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk23u/pull/63.diff">https://git.openjdk.org/jdk23u/pull/63.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk23u/pull/63#issuecomment-2283782477)

### CODE REVIEW

**Review by iwanowww — APPROVED** [review]

**Comment by theRealAph** [comment]
Thanks @iwanowww . Do I need another reviewer?

**Comment by vnkozlov** [comment]
> Thanks @iwanowww . Do I need another reviewer?

No, for clean backport.

**Comment by kevinrushforth** [comment]
> > Thanks @iwanowww . Do I need another reviewer?
> 
> No, for clean backport.

The review requirement has been met. Maintainer approval is pending.

**Pull request closed without merging**
